extensions/src/main/java/org/pentaho/platform/web/http/api/resources/RepositoryResource.java

```diff
@@ -650,6 +652,17 @@ private boolean hasOperationId( final List<IPluginOperation> operations, final S
 
   protected Response doService( String contextId, String resourceId ) throws ObjectFactoryException,
       PluginBeanException, IOException, URISyntaxException {
+    // Check if we are editing a content
+    if(resourceId != null && (resourceId.equals("editor") || resourceId.equals("edit"))) {
+       // Check if user has permission to edit the content
+       if(!canEdit()) {
+         logger.error( MessageFormat.format( "User is not allowed to edit the content", contextId ) );
+         return Response.status(Status.UNAUTHORIZED).build();
+       }
+    }
```

Comment:
I have found an “edge case”, where it is currently possible to bypass validation. We’re testing for a `resourceId` of `editor` or `edit`. However, one can currently also access the Analyzer editor using the `resourceId`  `xanalyzer.editor`, which is the content-generator's identifier, as registered in the PentahoSystem.

Should we test for the _presence_ of these special strings instead, with boundaries, e.g. `\beditor\b`?
Or, should we explicitly split the `resourceId` string and test the second part? I'd choose this one, given we do know its possible structures and this should introduce less surprises (unwanted behaviors).

Inescapably, looking at URLs for presumed special strings, and, especially, in the Content Generator realm, is risky business…

In a more sophisticated design, the specific Content Generator implementation being invoked would be annotated with a Java annotation stating its “edit” nature — opting in. The CG handler would look for this annotation and, when present, would enforce the corresponding permissions were held by the current user.

This is how I have designed the indication of necessity for CSRF control for CG classes.

However, this surely would take us longer to develop, and would not fit the currently available time-window.